src/cudamatrix/cu-matrix.cc

```diff
@@ -1309,11 +1309,22 @@ void CuMatrixBase<Real>::GroupMax(const CuMatrixBase<Real> &src) {
 #if HAVE_CUDA == 1
   if (CuDevice::Instantiate().Enabled()) {
     Timer tim;
-    dim3 dimBlock(CU2DBLOCK, CU2DBLOCK);
-    dim3 dimGrid(n_blocks(NumCols(), CU2DBLOCK),
-                 n_blocks(NumRows(), CU2DBLOCK));
-    cuda_group_max(dimGrid, dimBlock, this->data_, src.data_,
-                   this->Dim(), src.Stride(), group_size);
+    // One thread block per row.
+    // Use 2D block for small group size to simplify the calculation
+    // Use 1D block for large group size
+    //   group size: 1 2 3 4 5 6 7 .. 12 13 .. 24 25 .. 48 49 .. 96 97 .. 192 193 .. 384 385 ...
+    // ragular_size: 1 1 1 2 2 2 4 ..  4  8 ..  8 16 .. 16 32 .. 32 64 ..  64 128 .. 128 256 ...
+    int ragular_size = CU1DBLOCK;
```

Comment:
perhaps you mean 'regular_size'.  But even then, I don't understand why you're using this word at all- perhaps it's a mistranslation?  What is the concept behind it?